packages/editor-ui/src/views/SetupWorkflowFromTemplateView/useCredentialSetupState.ts

```diff
@@ -0,0 +1,227 @@
+import type { Ref } from 'vue';
+import { computed, ref } from 'vue';
+import type { ICredentialsResponse, INodeUi } from '@/Interface';
+import { useCredentialsStore } from '@/stores/credentials.store';
+import { useNodeTypesStore } from '@/stores/nodeTypes.store';
+import { getAppNameFromNodeName } from '@/utils/nodeTypesUtils';
+import type { TemplateCredentialKey } from '@/utils/templates/templateTransforms';
+import {
+	keyFromCredentialTypeAndName,
+	normalizeTemplateNodeCredentials,
+} from '@/utils/templates/templateTransforms';
+import type { INodeCredentialDescription, INodeCredentialsDetails } from 'n8n-workflow';
+import type { NodeTypeProvider } from '@/utils/nodeTypes/nodeTypeTransforms';
+import { getNodeTypeDisplayableCredentials } from '@/utils/nodes/nodeTransforms';
+import sortBy from 'lodash-es/sortBy';
+
+//#region Types
+
+export type NodeCredentials = {
```

Comment:
All of these have been moved (with slight modifications) from `setupTemplate.store.ts`